src/DotNetWorker.Core/FunctionsApplication.cs

```diff
@@ -61,12 +65,24 @@ public void LoadFunction(FunctionDefinition definition)
             _diagnostics.OnFunctionLoaded(definition);
         }
 
-        public Task InvokeFunctionAsync(FunctionContext context)
+        public async Task InvokeFunctionAsync(FunctionContext context)
         {
             var scope = new FunctionInvocationScope(context.FunctionDefinition.Name, context.InvocationId);
-            using (_logger.BeginScope(scope))
+
+            using var logScope = _logger.BeginScope(scope);
+            using Activity? invokeActivity = _functionActivitySourceFactory.StartInvoke(context);
+
+            try
             {
-                return _functionExecutionDelegate(context);
+                await _functionExecutionDelegate(context);
+                invokeActivity?.SetStatus(ActivityStatusCode.Ok);
```

Comment:
faas spec does mention status, but it's common in other conventions to leave status unset. This way if user sets error/ok status on current activity that finished without exception, we won't override it here

Reply:
so -- only set it to Error if we catch the Exception here? Otherwise leave it alone?

Reply:
yep!